releasenotes/notes/deprecate-legacy-code-fe9db8a009b6f5d2.yaml

```diff
@@ -0,0 +1,5 @@
+---
```

Comment:
Maybe this PR is not the place, but at some point I think a prelude section that outlines the new direction for the end user reading the release note and talks a bit about the deprecation and migration path. Maybe we can wait until we have a migration guide ready and have this aspect of the release note be part of that. This then may help a reader to understand the implications to what they currently have. I find the deprecation note as it stands here does not really help me too much - it says "all code that was migrated", so does this impact me or not reading this would be my question. (The answer is pretty much always going to be yes since pretty much all code was migrated!) With this text as-is, to try to figure anything, I have to go to 749 which references another PR (722) which lists a bunch too. Perhaps with an overall description of the direction/migration it will be clear to user and this will end up being a bit of detail that this unless you trace through the ref'd PR and try to work things out yourself it will not offer much help! Maybe we could alter this to something like the following which is a bit more descriptive. (And maybe later we can even update this release note to refer the user back to the prelude we add)

The vast majority of the existing Nature code has been migrated/refactored to `qiskit_nature/second_q` and the corresponding existing code has all been deprecated according to issue:....

Reply:
I absolutely agree! I was thinking that I could add one major release note at the end of the refactoring process, which explains the new workflow.

That being said, the contents of this particular release note could be improved. Maybe we can add a short overview of what has been deprecated in favor of where it goes? Maybe even in table format?
Maybe something with the following content in mind:
```
qiskit_nature.algorithms -> qiskit_nature.second_q.algorithms
qiskit_nature.algorithms.pes_samplers -> removed; refer to X for more details
qiskit_nature.circuit -> qiskit_nature.second_q.circuit
qiskit_nature.converters -> qiskit_nature.second_q.mappers
qiskit_nature.drivers -> qiskit_nature.second_q.drivers
qiskit_nature.mappers -> qiskit_nature.second_q.mappers
qiskit_nature.operators -> qiskit_nature.second_q.operators
qiskit_nature.problems -> qiskit_nature.second_q.problems
qiskit_nature.problems.sampling -> moved to prototypes; refer to X for more details
qiskit_nature.properties -> qiskit_nature.second_q.hamiltonians, qiskit_nature.second_q.properties (refer to X for more details)
qiskit_nature.results -> qiskit_nature.second_q.problems
qiskit_nature.transformers -> qiskit_nature.second_q.transformers
```